src/main/java/com/browserstack/automate/ci/jenkins/local/JenkinsBrowserStackLocal.java

```diff
@@ -51,7 +51,7 @@ public JenkinsBrowserStackLocal(String accesskey, LocalConfig localConfig, Strin
             arguments.add(args[i]);
         }
 
-        localIdentifier = UUID.randomUUID().toString().replaceAll("\\-", "") + "-" + buildTag.replaceAll("\\s","").replaceAll("#", "_");
+        localIdentifier = UUID.randomUUID().toString() + "-" + buildTag.replaceAll("[^\\w\\-\\.]", "_");
```

Comment:
Thanks for your pull request!

Correct me If I missed something here but shouldn't it be
```java
localIdentifier = UUID.randomUUID().toString() + "-" + buildTag;
localIdentifier = localIdentifier.replaceAll("[^\\w\\-\\.]", "_");
```
Asking this because I couldn't see `UUID.randomUUID().toString().replaceAll("\\-", "")` in latest commit. Meanwhile It's counterintuitive because we are appending `-` right after replacing it with blank string [here](https://github.com/jenkinsci/browserstack-integration-plugin/pull/13/files#diff-f5530028d62562c413d867f45daab6aaL54)

I was also wondering @grahamlyons,  do we even need `buildTag` appending to `localIdentifier`? can it be just UUID, since [UUID is unique](https://stackoverflow.com/questions/1155008/how-unique-is-uuid) space and time 😄

Reply:
@mohitmun, I think the build tag is there so that different local BrowserStack sessions can be traced if necessary. A UUID will certainly be unique but it doesn't give any indication about which job or build it's come from.

I don't have any strong feelings about replacing characters in the UUID or not. @damianszczepanik suggested not operating on the UUID as the docs indicate that it will only return valid characters.

I would be in favour of keeping the build tag appended but I don't really mind whether or not we call `replaceAll` on the UUID - as `-` is a valid character it won't be replaced anyway (but we could put back in the `replaceAll("\\-", "")` to keep the behaviour the same as the current release.

Reply:
@grahamlyons ah! this makes sense! we will get it merged and released